packages/cli/src/local-data/local-data.utils.ts

```diff
@@ -1,3 +1,122 @@
+  const logger = log.getLogger(METICULOUS_LOGGER_NAME);
+
+  await mkdir(dirname(filePath), { recursive: true });
+
+  // We create a lock file so that if multiple processes try downloading at the same
+  // time they don't interfere with each other. The second process to run will
+  // wait for the first process to complete, and then return straight away because
+  // it'll notice the file already exists.
+  const releaseLock = await waitToAcquireLockOnFile(filePath);
+
+  try {
+    const existingData = await readFile(filePath)
+      .then((data) => JSON.parse(data.toString("utf-8")))
+      .catch(() => null);
+    if (existingData) {
+      logger.debug(`Reading ${dataDescription} from local copy in ${filePath}`);
+      return existingData;
+    }
+
+    const downloadedData = await downloadJson();
+    if (downloadedData) {
+      await writeFile(filePath, JSON.stringify(downloadedData, null, 2));
+      logger.debug(`Wrote ${dataDescription} to ${filePath}`);
+    }
+    return downloadedData;
+  } finally {
+    await releaseLock();
+  }
+};
+
+const waitToAcquireLockOnFile = async (
+  filePath: string
+): Promise<ReleaseLock> => {
+  // In many cases the file doesn't exist yet, and can't exist yet (need to download the data, and creating an
+  // empty file beforehand is risky if the process crashes, and a second process tries reading the empty file).
+  // However proper-lockfile requires us to pass a file or directory as the first arg. This path is just used
+  // to detect if the same process tries taking out multiple locks on the same file. It just needs to be calculated
+  // as something that's unique to the file, and gives the same path for a given file everytime. So we create our
+  // own lock-target directory for this purpose (directory not file since mkdir is guaranteed to be synchronous).
+  // The path needs to actually exist, since proper-lockfile resolves symlinks on it.
+  const lockDirectory = `${filePath}.lock-target`;
+  await mkdir(lockDirectory, { recursive: true });
+
+  try {
+    const releaseLock = await lock(lockDirectory, {
+      retries: LOCK_RETRY_OPTIONS,
+      lockfilePath: `${filePath}.lock`,
+    });
+    return async () => {
+      // Clean up our directory _before_ releasing the lock
+      // Note: it may have already been deleted: if we're just coming out of
+      // a lock released by someone else then they will have already deleted
+      // the directory
+      await rm(lockDirectory, { recursive: true, force: true });
+      await releaseLock();
+    };
+  } catch (err) {
+    await rmdir(lockDirectory);
+    throw err;
+  }
+};
+
+export const fileExists = (filePath: string) =>
+  access(filePath)
+    .then(() => true)
+    .catch(() => false);
+
+export const waitToAcquireLockOnDirectory = (
+  directoryPath: string
+): Promise<ReleaseLock> => {
+  return lock(directoryPath, {
+    retries: LOCK_RETRY_OPTIONS,
+    lockfilePath: join(directoryPath, "dir.lock"),
```

Comment:
Not sure why the default is not suitable?

```suggestion
```

Reply:
This is what they recommend doing in their docs for directories (I guess it means if directory gets deleted so does the lock file). The default is designed for files.

Reply:
It does not read as a recommendation to me? The default behaviour works just fine for folders.